client/state/components/drop-zone/reducer.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+import {
+	DROPZONE_SHOW,
+	DROPZONE_HIDE
+} from 'state/action-types';
+
+import { createReducer } from 'state/utils';
+
+export default createReducer( {},
+	{
+		[ DROPZONE_SHOW ]: ( state = {} ) => {
+			return {
+				...state,
+				isVisible: true
+			};
+		},
+		[ DROPZONE_HIDE ]: ( state = {} ) => {
+			return {
+				...state,
+				isVisible: false
+			};
+		},
+	}
+);
```

Comment:
The way the reducer is built, a `DROPZONE_SHOW` action fired when the state already contains `isVisible: true` will trigger a state change anyway. Is that intentional?

If not, a low-cost way to address this (avoiding manual things like `state.isVisible ? state : { ...state, isVisible: true }`) is using `combineReducers` like so:

```js
import { stubFalse, stubTrue } from 'lodash';

const isVisible = createReducer( false, {
  [ DROPZONE_SHOW ]: stubTrue,
  [ DROPZONE_SHOW ]: stubFalse,
} );

export default combineReducers( { isVisible } );
```

Which makes sense: ultimately, your decomposing your state one step further, freeing yourself from managing objects and their references, letting Redux do its thing. The setting of `dropZone.isVisible` is independent from any potential keys `dropZone.*`.